panoply/datasource.py

```diff
@@ -7,9 +10,12 @@ class DataSource(events.Emitter):
     options = None
 
     # data source constructor
-    def __init__(self, source, options = {}):
-        super(DataSource, self).__init__()
-        
+    def __init__(self, source, options = {}, events = {}):
+        super(DataSource, self).__init__(events)
+
+        if 'destination' not in source:
```

Comment:
You are right, the SDK used the same source dictionary, hence after the super constructor, destination was always available.
I fixed it by making a copy of `source` in the super constructor and adding the destination to that copy. That was, `source` in the data source would still be unchanged ...